### PR TITLE
feat: Advanced markdown recipe import mode

### PR DESCRIPTION
**Pull request by basicallysteve**
## Summary

- Adds an **Advanced** toggle on `/recipes/new` alongside the existing Guided form
- Users write a structured markdown template (title, metadata, ingredients, steps) and parse it into a full Recipe in one step
- A **Recipe Import Preview** step shows ingredient resolution results before committing — matched foods get a green badge, partial matches show candidate buttons to pick from, unresolved ingredients get an inline search box or can be skipped
- Missing metadata (meal, serves, prep/cook time) appears as editable fallbacks in the preview
- Steps are converted from numbered plain text to `<p>` HTML, compatible with the existing `recipeSteps` schema
- After confirmation the recipe is created as a draft and the user lands on the recipe detail page

## What changed

**New files:**
- `src/utils/recipeMarkdownParser.ts` — client-side parser for the template format
- `src/utils/recipeLanguage.ts` — custom CodeMirror 6 `StreamParser` with section-aware highlighting (quantities, known units as keywords, unknown units as error tokens)
- `src/types/RecipeImport.ts` — shared `ResolvedIngredient` type
- `app/api/recipes/import/resolve-ingredients/route.ts` — server endpoint: case-insensitive food lookup, up to 3 candidates fallback
- `src/views/Recipe/MarkdownImport.tsx` — editor + preview component (three states: `editor` → `resolving` → `preview`)
- `src/views/Recipe/markdownImport.scss` — styles matching the app's VS Code-like dark theme
- `docs/adr/0017-markdown-import-split-parsing.md` — ADR for client/server parsing split
- `docs/adr/0018-codemirror-custom-recipe-language.md` — ADR for CodeMirror custom language choice

**Modified:**
- `src/views/Recipe/Store.tsx` — adds Guided / Advanced mode toggle to toolbar
- `src/views/Recipe/store.scss` — adds `.toolbar-mode-toggle` styles
- `CONTEXT.md` — three new glossary terms: Markdown Recipe Import, Recipe Import Preview, Ingredient Resolution
- `package.json` / `bun.lock` — adds `@uiw/react-codemirror` and `@codemirror/language`

## Tests

39 new tests across two files:
- `src/utils/recipeMarkdownParser.test.ts` — 20 unit tests for the parser (ingredient line formats, metadata parsing, edge cases)
- `src/views/Recipe/MarkdownImport.test.tsx` — 19 component tests (mode toggle, editor state, preview: matched/candidates/unresolved badges, skip/include, create recipe API calls)

## Reviewer notes

- CodeMirror is loaded via `next/dynamic` with `ssr: false` to keep it client-only
- The `@/` path alias resolves to `src/` — the API route type is kept in `src/types/RecipeImport.ts` so both the route handler and the client component can import it
- Ingredient resolution is intentionally split: markdown parsing on the client (no I/O), food lookup on the server (needs DB access)

🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### CODE REVIEW

File: src/views/Recipe/MarkdownImport.tsx [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+            onClick={() => { setMode('editor'); setError(null) }}
+          >
+            ← Edit Markdown
+          </button>
+          <button
+            type="button"
+            className="primary-button"
+            onClick={handleCreate}
+            disabled={submitting || !previewName.trim()}
+          >
```
[the remark this answers is not in the document]
Comment (basicallysteve) [reply src/views/Recipe/MarkdownImport.tsx:328]: Added `hasUnresolved` to the `disabled` prop on the Create Recipe button — it's blocked until every ingredient is either matched/selected or explicitly skipped.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/views/Recipe/MarkdownImport.tsx [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+          const unit = r.parsed.unit ?? food.servingUnit
+          const calories =
+            calculateCalories({
+              baseCalories: food.calories || 0,
+              baseServingSize: food.servingSize || 1,
+              baseServingUnit: food.servingUnit,
+              targetAmount: qty,
+              targetUnit: unit,
+            }) ?? 0
+          return { food, quantity: qty, servingUnit: unit, calories }
```
[the remark this answers is not in the document]
Comment (basicallysteve) [reply src/views/Recipe/MarkdownImport.tsx:139]: Now looks up `measurement?.gramsPerUnit` from `food.measurements` and passes it to `calculateCalories`, matching the pattern in `Index.tsx`.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/utils/recipeMarkdownParser.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+const INGREDIENT_PATTERN = /^-?\s*(\d+(?:[.,]\d+)?)\s*([a-zA-Z][a-zA-Z-]*)\s+(.+)$/
+
+export function parseIngredientLine(raw: string): ParsedIngredient {
+  const trimmed = raw.trim()
+  const match = trimmed.match(INGREDIENT_PATTERN)
+  if (!match) return { raw: trimmed, quantity: null, unit: null, foodName: null }
+
+  const quantity = parseFloat(match[1].replace(',', '.'))
+  const unit = match[2]
+  const foodName = match[3].trim()
```
[the remark this answers is not in the document]
Comment (basicallysteve) [reply src/utils/recipeMarkdownParser.ts:39]: Added a `CANONICAL_UNIT` map (lowercase → canonical) in `recipeMarkdownParser.ts`. Parsed units are now normalised at parse time (`tbs` → `Tbs`, `FL-OZ` → `fl-oz`, etc.) while truly unknown units pass through unchanged.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: app/api/recipes/import/resolve-ingredients/route.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+export type { ResolvedIngredient }
+
+export async function POST(request: Request) {
+  const body: { ingredients: ParsedIngredient[] } = await request.json()
+  const { ingredients } = body
+
+  if (!Array.isArray(ingredients)) {
+    return NextResponse.json({ error: 'Invalid input' }, { status: 400 })
+  }
```
[the remark this answers is not in the document]
Comment (basicallysteve) [reply app/api/recipes/import/resolve-ingredients/route.ts:26]: Added a per-item shape check before processing — returns 400 if any element is null, non-object, or missing a `raw` string property.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/views/Recipe/MarkdownImport.tsx [hunk 5]
```diff
@@ -0,0 +1,395 @@
+'use client'
```
Comment (basicallysteve) [inline src/views/Recipe/MarkdownImport.tsx:1]: .

Comment (basicallysteve) [inline src/views/Recipe/MarkdownImport.tsx:1]: .

Comment (basicallysteve) [inline src/views/Recipe/MarkdownImport.tsx:1]: .

Comment (basicallysteve) [inline src/views/Recipe/MarkdownImport.tsx:1]: .
